src/lsp.rs

```diff
@@ -26,22 +26,39 @@ use crate::{
     fixing::{get_newline_offsets_rope_or_slice, AccumulatedEdits},
     tree_sitter::{self, InputEdit, Parser, Point, Tree},
     tree_sitter_grep::{Parseable, SupportedLanguage},
-    Args, ArgsBuilder, FixingForSliceRunContext, FixingForSliceRunStatus, MutRopeOrSlice,
+    Args, ArgsBuilder, Config, FixingForSliceRunContext, FixingForSliceRunStatus, MutRopeOrSlice,
     PerConfigContext, SliceRunStatus, ViolationWithContext,
 };
 
 const APPLY_ALL_FIXES_COMMAND: &str = "tree-sitter-lint.applyAllFixes";
 
+pub enum ArgsOrConfig<'a> {
+    Args(Args),
+    Config(&'a Config),
+}
+
+impl<'a> From<Args> for ArgsOrConfig<'a> {
+    fn from(value: Args) -> Self {
+        Self::Args(value)
+    }
+}
+
+impl<'a> From<&'a Config> for ArgsOrConfig<'a> {
+    fn from(value: &'a Config) -> Self {
+        Self::Config(value)
+    }
+}
+
 pub trait LocalLinter: Send + Sync {
     fn run_for_slice<'a>(
         &self,
         file_contents: impl Into<RopeOrSlice<'a>>,
         tree: Option<Tree>,
         path: impl AsRef<Path>,
-        args: Args,
+        args_or_config: ArgsOrConfig,
         language: SupportedLanguage,
         per_config_context: Option<&PerConfigContext>,
-    ) -> SliceRunStatus;
+    ) -> (SliceRunStatus, Option<Config>);
```

Comment:
I thought about `Config` being a field on `PerConfigContext` but I think they need to be asymmetrical basically because the local binary needs to pass in an `&Config` every time to the `tree_sitter_lint` entry point (at least as it's currently structured, which I think makes sense?) so that didn't seem to work (at least easily) as far as passing around ownership and such